Description: Fix ftbfs on alpha, see #962676
From: Helmut Grohne <helmut@subdivi.de>
--- a/programs/util.c
+++ b/programs/util.c
@@ -34,7 +34,7 @@
 #else
 #  include <unistd.h>     /* chown, stat */
 #  include <sys/stat.h>   /* utimensat, st_mtime */
-#  if (PLATFORM_POSIX_VERSION >= 200809L && defined(st_mtime)) \
+#  if (PLATFORM_POSIX_VERSION >= 200809L && (defined(st_mtime) || defined(__alpha__))) \
       || (defined(__FreeBSD__) && __FreeBSD_version >= 1100056)
 #    define ZSTD_USE_UTIMENSAT 1
 #  else
