Description: Fix the build on kFreeBSD - no UTIME_NOW.
Forwarded: no
Author: Peter Pentchev <roam@ringlet.net>
Last-Update: 2024-06-13

--- a/programs/util.c
+++ b/programs/util.c
@@ -35,7 +35,7 @@
 #  include <unistd.h>     /* chown, stat */
 #  include <sys/stat.h>   /* utimensat, st_mtime */
 #  if (PLATFORM_POSIX_VERSION >= 200809L && (defined(st_mtime) || defined(__alpha__))) \
-      || (defined(__FreeBSD__) && __FreeBSD_version >= 1100056)
+      || (defined(__FreeBSD__) && __FreeBSD_version >= 1100056 && !defined(__FreeBSD_kernel__))
 #    define ZSTD_USE_UTIMENSAT 1
 #  else
 #    define ZSTD_USE_UTIMENSAT 0
