diff -u -r wireshark-1.10.14-clean/CMakeLists.txt wireshark-1.10.14/CMakeLists.txt
--- wireshark-1.10.14-clean/CMakeLists.txt	2015-05-12 23:21:07.000000000 +0300
+++ wireshark-1.10.14/CMakeLists.txt	2016-01-29 17:29:04.475933938 +0200
@@ -465,6 +465,7 @@
 		plugins/profinet
 		plugins/stats_tree
 		plugins/unistim
+		plugins/wavelog
 		plugins/wimax
 		plugins/wimaxasncp
 		plugins/wimaxmacphy
diff -u -r wireshark-1.10.14-clean/configure.ac wireshark-1.10.14/configure.ac
--- wireshark-1.10.14-clean/configure.ac	2015-05-12 23:21:08.000000000 +0300
+++ wireshark-1.10.14/configure.ac	2016-01-29 17:22:56.771925836 +0200
@@ -2715,6 +2715,7 @@
   plugins/profinet/Makefile
   plugins/stats_tree/Makefile
   plugins/unistim/Makefile
+  plugins/wavelog/Makefile
   plugins/wimax/Makefile
   plugins/wimaxasncp/Makefile
   plugins/wimaxmacphy/Makefile
diff -u -r wireshark-1.10.14-clean/epan/Makefile.am wireshark-1.10.14/epan/Makefile.am
--- wireshark-1.10.14-clean/epan/Makefile.am	2015-05-12 23:21:08.000000000 +0300
+++ wireshark-1.10.14/epan/Makefile.am	2016-01-29 17:25:48.875929628 +0200
@@ -226,6 +226,7 @@
 	../plugins/irda/packet-irda.c \
 	../plugins/m2m/packet-m2m.c \
 	../plugins/m2m/wimax_tlv.c \
+        ../plugins/wavelog/packet-wavelog.c \
         ../plugins/wimax/crc.c \
         ../plugins/wimax/crc_data.c \
         ../plugins/wimax/mac_hd_generic_decoder.c \
diff -u -r wireshark-1.10.14-clean/Makefile.am wireshark-1.10.14/Makefile.am
--- wireshark-1.10.14-clean/Makefile.am	2015-05-12 23:21:07.000000000 +0300
+++ wireshark-1.10.14/Makefile.am	2016-01-29 17:08:21.375906549 +0200
@@ -301,6 +301,7 @@
 	-dlopen plugins/profinet/profinet.la \
 	-dlopen plugins/stats_tree/stats_tree.la \
 	-dlopen plugins/unistim/unistim.la \
+	-dlopen plugins/wavelog/wavelog.la \
 	-dlopen plugins/wimax/wimax.la \
 	-dlopen plugins/wimaxmacphy/wimaxmacphy.la
 
diff -u -r wireshark-1.10.14-clean/packaging/nsis/Makefile.nmake wireshark-1.10.14/packaging/nsis/Makefile.nmake
--- wireshark-1.10.14-clean/packaging/nsis/Makefile.nmake	2015-05-12 23:21:12.000000000 +0300
+++ wireshark-1.10.14/packaging/nsis/Makefile.nmake	2016-01-29 17:37:06.255944553 +0200
@@ -56,6 +56,7 @@
 	../../plugins/profinet/profinet.dll \
 	../../plugins/stats_tree/stats_tree.dll \
 	../../plugins/unistim/unistim.dll \
+	../../plugins/wavelog/wavelog.dll \
 	../../plugins/wimax/wimax.dll \
 	../../plugins/wimaxasncp/wimaxasncp.dll \
 	../../plugins/wimaxmacphy/wimaxmacphy.dll \
diff -u -r wireshark-1.10.14-clean/packaging/nsis/wireshark.nsi wireshark-1.10.14/packaging/nsis/wireshark.nsi
--- wireshark-1.10.14-clean/packaging/nsis/wireshark.nsi	2015-05-12 23:21:12.000000000 +0300
+++ wireshark-1.10.14/packaging/nsis/wireshark.nsi	2016-01-29 17:39:07.547947225 +0200
@@ -851,6 +851,7 @@
 File "${STAGING_DIR}\plugins\${VERSION}\opcua.dll"
 File "${STAGING_DIR}\plugins\${VERSION}\profinet.dll"
 File "${STAGING_DIR}\plugins\${VERSION}\unistim.dll"
+File "${STAGING_DIR}\plugins\${VERSION}\wavelog.dll"
 File "${STAGING_DIR}\plugins\${VERSION}\wimax.dll"
 File "${STAGING_DIR}\plugins\${VERSION}\wimaxasncp.dll"
 File "${STAGING_DIR}\plugins\${VERSION}\wimaxmacphy.dll"
diff -u -r wireshark-1.10.14-clean/plugins/Makefile.am wireshark-1.10.14/plugins/Makefile.am
--- wireshark-1.10.14-clean/plugins/Makefile.am	2015-05-12 23:21:12.000000000 +0300
+++ wireshark-1.10.14/plugins/Makefile.am	2016-01-29 17:00:48.079896561 +0200
@@ -35,6 +35,7 @@
 	profinet \
 	stats_tree \
 	unistim \
+	wavelog \
 	wimax \
 	wimaxasncp \
 	wimaxmacphy
diff -u -r wireshark-1.10.14-clean/plugins/Makefile.nmake wireshark-1.10.14/plugins/Makefile.nmake
--- wireshark-1.10.14-clean/plugins/Makefile.nmake	2015-05-12 23:21:12.000000000 +0300
+++ wireshark-1.10.14/plugins/Makefile.nmake	2016-01-29 17:03:16.291899827 +0200
@@ -18,6 +18,7 @@
 	profinet    \
 	stats_tree  \
 	unistim     \
+	wavelog     \
 	wimax       \
 	wimaxasncp  \
 	wimaxmacphy
